### ISSUE TITLE
unpack without bounds check stops ServiceBrowser

### ISSUE THREAD

**Issue by gbiddison** [jstasiak/python-zeroconf#65]
This exception is occuring at intervals since 0.17.6; stops the browser thread. Exception always occurs in a pair.
May be a bad packet, since the unpack looks to be unpacking based on packet information.

```python
Exception in thread zeroconf-Engine:
Traceback (most recent call last):
  File "//anaconda/envs/py34/lib/python3.4/threading.py", line 911, in _bootstrap_inner
    self.run()
  File "/Users/gbiddison/source/sygo/tools/BeagleDiff/zeroconf.py", line 957, in run
    reader.handle_read(socket_)
  File "/Users/gbiddison/source/sygo/tools/BeagleDiff/zeroconf.py", line 994, in handle_read
    msg = DNSIncoming(data)
  File "/Users/gbiddison/source/sygo/tools/BeagleDiff/zeroconf.py", line 569, in __init__
    self.read_others()
  File "/Users/gbiddison/source/sygo/tools/BeagleDiff/zeroconf.py", line 618, in read_others
    type_, class_, ttl, length = self.unpack(b'!HHiH')
  File "/Users/gbiddison/source/sygo/tools/BeagleDiff/zeroconf.py", line 574, in unpack
    format_, self.data[self.offset:self.offset + length])
struct.error: unpack requires a bytes object of length 10
```

**Comment by stephenrauch** [comment]
Before to 0.17.6 there was an blanket exception handler around:

```python
self.readers[socket_].handle_read(socket_)
```
in the Engine.run().  Will need to put something back in around the DNSIncoming() init (unpacking)

**Comment by stephenrauch** [comment]
@gbiddison 

I believe I have addressed this issue via #68.  If you could test it in your environment and report back that would be greatly appreciated.

**Comment by gbiddison** [comment]
:shipit: Looks great.
